limits.go

```diff
@@ -28,8 +28,8 @@ const (
 )
 
 type limitError struct {
-	Name string
-	Type schemaType
+	Name string     `json:"path"`
+	Type schemaType `json:"type"`
```

Comment:
Why return numeric schema type values to users?

Reply:
Fixed - I had to add `MarshalJSON` functions for all errors as a part of resolving merge conflicts with #6, and the schema type is now output as a string instead.